apps/smart-contracts/core/test/TokenSender.test.ts

```diff
@@ -67,6 +77,42 @@ describe('=> TokenSender', () => {
         `AccessControl: account ${user.address.toLowerCase()} is missing role ${await tokenSender.SET_PRICE_ROLE()}`
       )
     })
+
+    it('sets to non-zero address', async () => {
+      await tokenSender.setPrice(user.address)
+
+      expect(await tokenSender.getPrice()).to.eq(user.address)
+    })
+
+    it('sets to zero address', async () => {
+      await tokenSender.setPrice(ZERO_ADDRESS)
+
+      expect(await tokenSender.getPrice()).to.eq(ZERO_ADDRESS)
+    })
+
+    it('is idempotent for non-zero address', async () => {
+      await tokenSender.setPrice(user.address)
+      expect(await tokenSender.getPrice()).to.eq(user.address)
+
+      await tokenSender.setPrice(user.address)
+
+      expect(await tokenSender.getPrice()).to.eq(user.address)
+    })
+
+    it('is idempotent for zero address', async () => {
+      await tokenSender.setPrice(ZERO_ADDRESS)
+      expect(await tokenSender.getPrice()).to.eq(ZERO_ADDRESS)
+
+      await tokenSender.setPrice(ZERO_ADDRESS)
+
+      expect(await tokenSender.getPrice()).to.eq(ZERO_ADDRESS)
+    })
+
+    it('emits event', async () => {
```

Comment:
nit:
```suggestion
    it('emits PriceChange', async () => {
```

same for other tests